src/lib/core/WeaveMessageLayer.cpp

```diff
@@ -497,16 +547,27 @@ bool WeaveMessageLayer::IsSendErrorNonCritical (WEAVE_ERROR err)
  *  @retval  errors generated from the lower Inet layer UDP endpoint during sending.
  *
  */
-WEAVE_ERROR WeaveMessageLayer::SendMessage(const IPAddress &destAddr, uint16_t destPort, InterfaceId sendIntfId,
-                                           PacketBuffer *payload, uint16_t msgSendFlags)
+WEAVE_ERROR WeaveMessageLayer::SendMessage(const IPAddress & destAddr, uint16_t destPort, InterfaceId sendIntfId,
```

Comment:
This is getting to be a monster of a method call. Any benefit, beyond complexity management such as code size, to refactoring it into a singular public method with supporting private methods?

Reply:
My impression is that the new method is of comparable complexity to the old one, but easier to follow due to the clear phasing of the procedure.  Nonetheless, it seems reasonable to factor out some of the logic, such as the code that selects the sending end point.  

I am loath, however, to expose the sendAction enum outside of this method, as it exists solely to make the sending logic easier to understand.  Thus I think I'll keep that part of the code within SendMessage().